mldsa/poly.c

```diff
@@ -323,16 +323,20 @@ __contract__(
  *           - Simplified from reference by removing buffer tail handling
  *             since buflen % 3 = 0 always holds true (STREAM128_BLOCKBYTES =
  *             168).
- *           - Modified rej_uniform interface to track offset directly. */
-void poly_uniform(poly *a, const uint8_t seed[MLDSA_SEEDBYTES], uint16_t nonce)
+ *           - Modified rej_uniform interface to track offset directly.
+ *           - Pass nonce packed in the extended seed array instead of a third
+ *             argument.
+ * */
+void poly_uniform(poly *a, const uint8_t seed[MLDSA_SEEDBYTES + 2])
 {
   unsigned int ctr;
   unsigned int buflen = POLY_UNIFORM_NBLOCKS * STREAM128_BLOCKBYTES;
   uint8_t buf[POLY_UNIFORM_NBLOCKS * STREAM128_BLOCKBYTES];
```

Comment:
While at it, can you add an alignment directive here? Otherwise, the performance may vary based on where the compiler puts this.

Reply:
done